### PR TITLE
Use a function to execute sudo

### PR DESCRIPTION
**Pull request by szepeviktor**
Functions are for doing things.
Variables are for storing values.

### CODE REVIEW

File: install.sh [hunk 1]
```diff
@@ -26,17 +29,19 @@ version_ge() {
...
 	version_ge "$(major_minor "${python_version_output##* }")" "$(major_minor "${REQUIRED_PYTHON_VERSION}")"
 }
 
+/usr/bin/sudo -l mkdir &>/dev/null || abort "This installer requires root or sudo."
+
+OS="$(uname)"
+if [[ "${OS}" != "Linux" ]] && [[ "${OS}" != "Darwin" ]]; then
+	abort "Sail CLI is only supported on macOS and Linux. Windows users can use Sail via WSL."
+fi
+
```
Comment (szepeviktor) [inline install.sh:44]: Move commands **after** function definitions.

File: install.sh [hunk 2]
```diff
@@ -9,11 +9,14 @@ abort() {
...
-	sudo -l mkdir &>/dev/null || abort "This installer requires root or sudo."
-	SUDO="sudo "
-fi
+sudo() {
+	if [ "$UID" -eq 0 ]; then
+ 		command "${@}"
+		return
+	fi
+
+ 	/usr/bin/sudo "${@}"
```
Comment (kovshenin) [inline install.sh:18]: We're making an assumption here that sudo is in /usr/bin/sudo, but it could be in a different place.

File: install.sh [hunk 3]
```diff
@@ -26,17 +29,19 @@ version_ge() {
...
-	abort "Sail CLI is only supported on macOS and Linux. Windows users can use Sail via WSL."
-fi
-
 test_python() {
 	local python_version_output
 	python_version_output="$("$1" --version 2>/dev/null)"
 	version_ge "$(major_minor "${python_version_output##* }")" "$(major_minor "${REQUIRED_PYTHON_VERSION}")"
 }
 
+/usr/bin/sudo -l mkdir &>/dev/null || abort "This installer requires root or sudo."
```
Comment (kovshenin) [inline install.sh:38]: Again, assumption of sudo location, as well as its existence. If the script is run as root and sudo is not available we should still be ok.

File: install.sh (hunk 2 above)
Comment (szepeviktor) [reply install.sh]: Resolved 🍏

File: install.sh (hunk 3 above)
Comment (szepeviktor) [reply install.sh]: Should I wrap this in `if [ ! "$UID" -eq 0 ]; then` too?

Comment (kovshenin) [reply install.sh]: I guess so, unless you can think of a better way to do it.

Comment (szepeviktor) [reply install.sh]: Done 🍏

**Comment by szepeviktor** [comment]
Thank you.

**Comment by kovshenin** [comment]
No, thank *you* @szepeviktor! :)

**Pull request merged**